src/material/datepicker/calendar-body.ts

```diff
@@ -216,10 +218,14 @@ export class MatCalendarBody implements OnChanges, OnDestroy, AfterViewChecked {
     return cellNumber == this.activeCell;
   }
 
-  /** Focuses the active cell after the microtask queue is empty. */
+  /**
+   * Focuses the active cell after the microtask queue is empty.
+   *
+   * Adds a 20ms delay to fix Voiceover losing focus when pressing PageUp/PageDown (issue #24330).
+   */
   _focusActiveCell(movePreview = true) {
     this._ngZone.runOutsideAngular(() => {
-      this._ngZone.onStable.pipe(take(1)).subscribe(() => {
+      this._ngZone.onStable.pipe(take(1), delay(FOCUS_ACTIVE_CELL_DELAY)).subscribe(() => {
```

Comment:
Add a comment here that explains the purpose of the `delay`

Reply:
There's an explanation for the `delay` a few lines above, which is right before the declaration of `_focusActiveCell` :)